### ISSUE TITLE
Why is optimizer momentum reset every 20 epochs on CIFAR10 2D examples?

### ISSUE THREAD

**Issue by vivianwhite** [kymatio/kymatio#1021]
Hello! I have been adapting the Kymatio 2D classification examples for a research project about randomized wavelet scattering. I noticed that in the CIFAR10 examples, during the training loop, the optimizer is being reset every 20 epochs. Links to code [here](https://github.com/kymatio/kymatio/blob/main/examples/2d/cifar_torch.py#L169), [here](https://github.com/kymatio/kymatio/blob/main/examples/2d/cifar_small_sample.py#L232), and [here](https://github.com/kymatio/kymatio/blob/main/examples/2d/cifar_resnet_torch.py#L198).
```
if epoch%20==0:
    optimizer = torch.optim.SGD(model.parameters(), lr=lr, momentum=0.9,
                                         weight_decay=0.0005)
```
The intention of this appears to be to update the learning rate, but it is also resetting the momentum buffer, which keeps a running history of the gradients. Is this intended behavior?

I looked into Scattering Networks for Hybrid Representation Learning by Edouard Oyallon (@edouardoyallon) and Eugene Belilovsky (@eugenium) and there seemed to be no explicit mention of resetting momentum buffers, though there was mention of a learning rate scheduler (paper linked [here](https://arxiv.org/pdf/1809.06367.pdf)). It appears that the paper was using a similar training process/hyperparameters to the Wide Resnet paper (linked [here](https://arxiv.org/abs/1605.07146)). In the Wide Resnet codebase, the optimizer momentum buffer also appears to be reset every time the learning rate scheduler updates (link to code [here](https://github.com/szagoruyko/wide-residual-networks/blob/master/pytorch/main.py#L164)). However, there is no explicit mention of restarting the momentum buffers in the paper.

**Comment by MuawizChaudhary** [comment]
I talked to @edouardoyallon and he indicated that the proper behavior should be to initialize the optimizer outside of the loop, and then decay learning rate every 20 epochs . 

You can implement this in one of two ways; 

1. you access the optimiser `param_group` then access and change the learning rate within the `param_group` or 
2. you define a [learning rate schedule](https://pytorch.org/docs/stable/generated/torch.optim.lr_scheduler.StepLR.html#torch.optim.lr_scheduler.StepLR) that updates every 20 epochs up to 80.

I'll leave it to you and/or others to choose which implementation you'll do.

**Comment by MuawizChaudhary** [comment]
I did find this interesting paper on the topic of restarting your momentum buffers. It's done for non-Neural Network models (like linear regression or some `log-sum-exp` functions). It seems like if done often enough, it can help the optimization procedure.  I've heard about stuff like effective learning rate, and how its something to take into consideration when you increase or decrease the size of batches when you finetune your model, among other hyper-parameters. I wonder if there's some connections between resetting momentum and effective learning rate, and a systematic study could be interesting here. 

But thats beyond the scope of this PR :)

**Comment by eickenberg** [comment]
it was probably meant to just update the LR - PR welcome

**Comment by vivianwhite** [comment]
I've added a PR (#1022) to resolve this in one of the example scripts, if everyone agrees I will add it to the other two scripts. :)
